loadgen/agent-trade-amm.js

```diff
@@ -0,0 +1,141 @@
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { amountMath } from '@agoric/ertp';
+import { pursePetnames, issuerPetnames } from './petnames';
+import { disp } from './display';
+import { allValues } from './allValues';
+
+// This is loaded by the spawner into a new 'spawned' vat on the solo node.
+// The default export function is called with some args.
+
+export default async function startAgent([key, home]) {
+  const { zoe, scratch, agoricNames, wallet } = home;
+
+  console.log(`trade-amm: building tools`);
+  // const runIssuer = await E(agoricNames).lookup('issuer', issuerPetnames.RUN);
+  const { runBrand, bldBrand, autoswap, runPurse, bldPurse } = await allValues({
+    runBrand: E(agoricNames).lookup('brand', issuerPetnames.RUN),
+    // bldBrand: E(agoricNames).lookup('brand', issuerPetnames.BLD),
+    bldBrand: E(E(wallet).getIssuer(issuerPetnames.BLD)).getBrand(),
+    autoswap: E(agoricNames).lookup('instance', 'autoswap'),
+    runPurse: E(wallet).getPurse(pursePetnames.RUN),
+    bldPurse: E(wallet).getPurse(pursePetnames.BLD),
+  });
+  // const bldBrand = await E(bldPurse).getAllegedBrand();
+  const publicFacet = await E(zoe).getPublicFacet(autoswap);
+
+  console.log(`trade-amm: tools installed`);
+
+  async function getBalance(which) {
+    let bal;
+    if (which === 'RUN') {
+      bal = await E(runPurse).getCurrentAmount();
+      if (bal.value === BigInt(0)) {
+        // some chain setups currently fail to make the purses visible with
+        // the right denominations
+        throw Error(`no RUN, trade-amm cannot proceed`);
+      }
+      return bal;
+    }
+    if (which === 'BLD') {
+      bal = await E(bldPurse).getCurrentAmount();
+      if (bal.value === BigInt(0)) {
+        throw Error(`no BLD, trade-amm cannot proceed`);
+      }
+      return bal;
+    }
+    throw Error(`unknown type ${which}`);
+  }
+
+  async function getBalances() {
+    return allValues({ run: getBalance('RUN'), bld: getBalance('BLD') });
+  }
+
+  async function buyRunWithBld(bldOffered) {
+    const proposal = harden({
+      want: { Out: amountMath.make(runBrand, BigInt(0)) },
```

Comment:
```suggestion
      want: { Out: amountMath.makeEmpty(runBrand, AssetKind.NAT) },
```
Could do this instead, but works both ways.